packages/hadron-oauth/README.md

```diff
@@ -0,0 +1,218 @@
+## Installation
+
+`npm install --save @brainhubeu/hadron-oauth`
+
+## Overview
+
+`hadron-oauth` is a Hadron utility package to simplify working with OAuth providers, such as Google and Facebook. It provides several utility functions that you can use to make writing OAuth2 authentication quicker and more streamlined.
+
+## Tutorial
+
+### Understanding OAuth flow
+
+The plan for our authentication flow is as follows:
+
+* The client makes a `GET` request to `/auth/{provider}` to begin the process of authentication.
+* The server redirects the client to a provider consent site.
+* The client receives an auth code.
+* The client makes a `POST` request to `/auth/{provider}/token` to exchange the auth code for an access token.
+
+We can then use the access token to make further request to the provider's API in order to fetch data about the user (such as their name or email).
+
+### Configuration
+
+We will need to provide certain information to `hadron-oauth` before we can proceed with the auth process. This information exists in Hadron's config file.
+
+```js
+// oauth.js
+module.exports = {
+  google: {
+    clientID: 'keyboard-cat',
+    clientSecret: 'shhh',
+    scope: ['https://www.googleapis.com/auth/userinfo.profile'],
+    redirectUri: 'http://localhost:8081/',
+  },
+};
+```
+
+The `clientID` and `clientSecret` fields need to be the same as these given to you by your provider (such as the Google API Console). The `redirectUri` must be exactly the same as the one given to your provider.
+
+`scope` is an array of strings defining the scopes your app requires from the user. See the [Google scopes](https://developers.google.com/identity/protocols/googlescopes) and [Facebook scopes](https://developers.facebook.com/docs/facebook-login/permissions/) for details. These can be used later to retrieve relevant data from your provider's APIs about the user.
+
+In this case, we'll also pretend that there is a dev server for React or Vue at `http://localhost:8081`, however we can just as well redirect the calls back to our own server.
```

Comment:
Are we limited to have an app in React/Vue? I would replace `React or Vue` with something like `Frontend App`.